spread.yaml

```diff
@@ -55,6 +55,29 @@ backends:
                 username: ubuntu
                 password: ubuntu
 
+    adhoc-ubuntu-core:
+        type: adhoc
+        environment:
+            SPREAD_ADHOC_UC_ADDRESS: "$(HOST: echo ${SPREAD_ADHOC_UC_ADDRESS:-localhost})"
+            SPREAD_ADHOC_UC_PORT: "$(HOST: echo ${SPREAD_ADHOC_UC_PORT:-8022})"
+        allocate: |
+            echo "Allocating $SPREAD_SYSTEM..."
+            ADDRESS $SPREAD_ADHOC_UC_ADDRESS:$SPREAD_ADHOC_UC_PORT
+        discard: |
+            echo "Discarding ad-hoc $SPREAD_SYSTEM"
+        restore: |
+            rm -rf $GOPATH/src
```

Comment:
Spread has been improved to support reuse properly and nicely with this case (and every other, actually). Please let me know how it goes @fgimenez.